courses/models.py

```diff
@@ -428,6 +437,11 @@ def available_runs(self, user):
             ).values_list("run__id", flat=True)
         return [run for run in self.unexpired_runs if run.id not in enrolled_runs]
 
+    @property
+    def marketing_url(self):
+        """Return the marketing URL for this course"""
+        return getattr(self.first_unexpired_run, "external_marketing_url", None)
```

Comment:
Same happens for the course as well when there is no external_marketing_url.

Reply:
Thanks for catching this, We should return `""` I'll do that in my next commit .